azurelinuxagent/common/utils/fileutil.py

```diff
@@ -76,6 +76,9 @@ def write_file(filepath, contents, asbin=False, encoding='utf-8', append=False):
     """
     Write 'contents' to 'filepath'.
     """
+    if contents is None:
```

Comment:
Are you _really_ sure this is what you want? If contents==None, then the file won't even be created. I would think that write_file(path, None) would create an empty file (file exists but is of zero length). That's the existing behavior, isn't it? I would expect this to break something.

Reply:
From our office conversation: the post-condition for this method should be "In the absence of a raised exception, a file named 'filepath' will exist on disk". I don't care if "contents==None => Exception" or "contents==None  => create empty file"; either one is acceptable.